### ISSUE TITLE
User icon hamburger menu cannot be easily closed with keyboard.

### ISSUE THREAD

**Issue by bick-jp** [internetarchive/openlibrary#7917]
It is not intuitive to close the user icon hamburger menu when navigating it with the keyboard. It should close when the focus is out of the menu and also when pressing the Esc key.

### Evidence / Screenshot (if possible)
The keyboard focus is now on the user icon hamburger menu.
![Screen Shot 2023-05-31 at 5 10 03 PM](https://github.com/internetarchive/openlibrary/assets/8709562/21b68938-8136-4954-bc60-825b9ceeb2fc)

Click the space key, then you can go inside the menu.
![Screen Shot 2023-05-31 at 5 10 19 PM](https://github.com/internetarchive/openlibrary/assets/8709562/cf3cf345-0f13-4833-8a1e-1735e6c34018)

But you cannot easily close this menu. After pressing the tab key many times, the focus goes out from the menu. But the menu keeps visible.

### Relevant url?

https://openlibrary.org/

### Steps to Reproduce

1. Go to https://openlibrary.org/
2. Use the tab key and go to the user icon hamburger menu, and click the space key to go inside the menu.

* Actual: The menu keeps visible even after the keyboard focus moves out from the menu.
* Expected: When users are inside the menu, they should be able to escape from the menu by Esc key. Also, it should automatically close when the keyboard focus moves out from the menu.

### Details

- **Logged in (Y/N)?** Y
- **Browser type/version?** Chrome Version 113.0.5672.126 (Official Build) (arm64)
- **Operating system?** macOS Monterey Version 12.3.1
- **Environment (prod/dev/local)?** prod

### Proposal & Constraints

Keyboard navigation of the [Microfost Accessibility page](https://www.microsoft.com/en-us/accessibility) might be a good reference.

**Comment by karlitto01** [comment]
Hi. I am new to this community and would like to contribute to this issue, if possible. Thanks!

**Comment by Abhishektharu** [comment]
can you Assign this to me??

**Comment by cdrini** [comment]
This issue is already being worked on by another contributor; see the good first issue label for other issues 👍

**Comment by cdrini** [comment]
Cc @TanyaPina who's working on it 👍

**Comment by Abhishektharu** [comment]
@cdrini can i be included too ? I know decent amount of javascript I would be glad to contribute in this PR.
